### PR TITLE
New Commit

### PR DESCRIPTION
**Pull request by frknasir**
Made some few changes. On the issue of deprecation, found some work arounds.

### CODE REVIEW

**Comment by rajdeol** [comment]
Thanks for your updates @frknasir , your contribution and addition to keep the code up to date are highly appreciated, however there are few things that you need to change to make this pull request acceptable.

I have added the comments in the commit, please make those changes. Once again I am really glad that you took the time and submitted a PR

File: Android/ApiAccess.java [hunk 1]
```diff
@@ -20,49 +24,158 @@
 import javax.crypto.Mac;
 import javax.crypto.spec.SecretKeySpec;
 
-/**
```
Comment (rajdeol) [inline Android/ApiAccess.java]: Do not remove the comments, you can change or add more information if you think the information is less or not relevant

File: Android/ApiAccess.java [hunk 2]
```diff
@@ -20,49 +24,158 @@
...
-    
+
     // header name in which we wil send security token
     protected final String API_SIGNATURE_HEADER = "X-HASH";
-    
+
     // header name in which we will send Unix time stamp
     protected final String TIME_STAMP_HEADER = "API-REQUEST-TIME";
 
-    /**
```
Comment (rajdeol) [inline Android/ApiAccess.java]: Do not remove the comments, you can change or add more information if you think the information is less or not relevant

File: Android/ApiAccess.java [hunk 3]
```diff
@@ -20,49 +24,158 @@
...
-     * constructor to set default values
-     */
-    public ApiAccess(){
+    String charset = "UTF-8";
+    HttpURLConnection conn;
+    DataOutputStream wr;
+    StringBuilder result;
+    URL urlObj;
+    JSONObject jObj = null;
+    StringBuilder sbParams;
```
Comment (rajdeol) [inline Android/ApiAccess.java]: combine the declaration of same type variables:
`StringBuilder result, sbParams;`

File: Android/ApiAccess.java [hunk 4]
```diff
@@ -20,49 +24,158 @@
...
+
     // add your API Secret in below variable
     protected final String api_secret = "Your-API-SECRET";
-    
-    // add your websrvice/server URL
-    protected final String api_url = "Path-OF-Your-API-SERVER";
+
     protected String time_stamp;
     protected static final String HEXES = "0123456789abcdef";
-    protected static final int DEFAULT_TIMEOUT = 120 * 1000;
```
Comment (rajdeol) [inline Android/ApiAccess.java]: Do not delete this and use it in `makeHttpRequest` function

File: Android/ApiAccess.java [hunk 5]
```diff
@@ -20,49 +24,158 @@
...
+                conn.setRequestProperty("Accept-Charset", charset);
+
+                while(iterator.hasNext()) {
+                    String key = (String) iterator.next();
+                    String value = (String) headers.get(key);
+                    conn.setRequestProperty(key, value);
+                }
+
+                conn.setReadTimeout(10000);
+                conn.setConnectTimeout(15000);
```
Comment (rajdeol) [inline Android/ApiAccess.java]: use `DEFAULT_TIMEOUT` constant here

File: Android/ApiAccess.java [hunk 6]
```diff
@@ -20,49 +24,158 @@
...
+
+                conn.setRequestProperty("Accept-Charset", charset);
+
+                while(iterator.hasNext()) {
+                    String key = (String) iterator.next();
+                    String value = (String) headers.get(key);
+                    conn.setRequestProperty(key, value);
+                }
+
+                conn.setReadTimeout(10000);
```
Comment (rajdeol) [inline Android/ApiAccess.java]: create `DEFAULT_READ_TIMEOUT` constant and use it here

File: Android/ApiAccess.java [hunk 7]
```diff
@@ -20,49 +24,158 @@
...
+
+                conn.setRequestProperty("Accept-Charset", charset);
+
+                while(iterator.hasNext()) {
+                    String key = (String) iterator.next();
+                    String value = (String) headers.get(key);
+                    conn.setRequestProperty(key, value);
+                }
+
+                conn.setConnectTimeout(15000);
```
Comment (rajdeol) [inline Android/ApiAccess.java]: use `DEFAULT_TIMEOUT` constant here

File: Android/TestActivity.java [hunk 8]
```diff
@@ -0,0 +1,110 @@
+import android.app.ProgressDialog;
```
Comment (rajdeol) [inline Android/TestActivity.java:1]: This file should have just one method `testApiCall` to show simple call to API
`ApiAccess apiAccess = new ApiAccess();`
`String URL = "Url-to-your-server";`
`HashMap<String, String> params = new HashMap<>();`
`params.put("test", "testvalue");`

`try {`
`               json = apiAccess.makeHttpRequest( URL, "POST", params);`
`} catch(Exception e) {`
`               e.printStackTrace();`
`}`

File: Android/layout/activity_test.xml [hunk 9]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (rajdeol) [inline Android/layout/activity_test.xml]: No need to add a layout file

**Comment by frknasir** [comment]
Hello bro...Ive made all the changes you suggested. Thank You.

**Comment by rajdeol** [comment]
Thanks for the changes and your contribution @frknasir  really appreciate it.

**Pull request merged**